apps/ledger-live-desktop/src/newArch/features/MemoTag/__tests__/MemoTagField.test.tsx

```diff
@@ -0,0 +1,54 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import React from "react";
+import { render, screen, fireEvent } from "tests/testUtils";
+import MemoTagField from "../components/MemoTagField";
+
+jest.mock("react-i18next", () => ({
+  ...jest.requireActual("react-i18next"),
+  useTranslation: () => ({
+    t: (key: string) => key,
+  }),
+}));
+
+describe("MemoTagField", () => {
+  it("renders MemoTagField with label and text field", () => {
+    render(<MemoTagField showLabel={true} />);
+    expect(screen.getByText("MemoTagField.label")).toBeInTheDocument();
+    expect(screen.getByRole("textbox")).toBeInTheDocument();
+    expect(screen.getByPlaceholderText("MemoTagField.placeholder")).toBeInTheDocument();
+  });
+
+  it("should render MemoTagField without label", () => {
+    render(<MemoTagField showLabel={false} />);
+    expect(screen.queryByText("MemoTagField.label")).not.toBeInTheDocument();
+  });
+
+  it("should call onChange when input value changes", () => {
+    const handleChange = jest.fn();
+    render(<MemoTagField onChange={handleChange} />);
+    fireEvent.change(screen.getByPlaceholderText("MemoTagField.placeholder"), {
+      target: { value: "new memo" },
+    });
+    expect(handleChange).toHaveBeenCalledTimes(1);
+  });
+
+  it("should render CaracterCountComponent if provided", () => {
+    const CaracterCountComponent = () => <div>Character Count</div>;
+    render(<MemoTagField CaracterCountComponent={CaracterCountComponent} />);
+    expect(screen.getByText("Character Count")).toBeInTheDocument();
+  });
+
+  it("should render instruction text on autoFocus", () => {
+    render(<MemoTagField autoFocus />);
+    expect(screen.getByPlaceholderText("MemoTagField.placeholder")).toHaveFocus();
+    expect(screen.getByText("MemoTagField.instruction")).toBeInTheDocument();
+  });
+
+  it("should render with error", () => {
+    render(<MemoTagField error={new Error("Error message")} />);
+    expect(screen.getByTestId("input-error")).toBeInTheDocument();
```

Comment:
Is there a reason to mock i18n and test keys instead of text displayed in English, please?

Reply:
nice catch

Reply:
Well there is no imperative reason to mock i18n, but at the same time your source of truth is the translation keys not the mapped value while it may be changed anytime. in this tests assertion, we make sure that the component renders with the right keys no matter what the mapped value is. If you want to push the test for mapped value per language then we can get rid of i18n mock , but make assertions on all expected translation wording. Which is overkill.

Reply:
I think that we need to keep consistent with other tests and test what is displayed to the user in English. Following the philosophy of the testing library https://testing-library.com/docs/example-react-intl/#translated-components-testing-stategy

Reply:
ok no problem, fixed ✅